keras_nlp/models/bloom/bloom_attention.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+
+from keras_nlp.backend import keras
+from keras_nlp.backend import ops
+
+
+class BloomAttention(keras.layers.Layer):
+    def __init__(
+        self,
+        num_heads,
+        dropout=0.0,
+        kernel_initializer="glorot_uniform",
+        bias_initializer="zeros",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.num_heads = num_heads
+        self.dropout = dropout
+        self.kernel_initializer = kernel_initializer
+        self.bias_initializer = bias_initializer
+
+    def build(self, inputs_shape):
+        _, seq_length, hidden_dim = inputs_shape
+
+        self.head_dim = hidden_dim // self.num_heads
+
+        # Layer-wise attention scaling
+        self.inv_norm_factor = 1.0 / math.sqrt(self.head_dim)
+
+        self._query_key_value_dense = keras.layers.Dense(
```

Comment:
Do we need a fused query, key, value here? Would it be possible to chop the weights during checkpoint conversion so that each projection gets it's own dense layer?

We are working on adding lora/quantization and similar to the Keras ecosystem right now. In that case we would benefit from split layers so we can target each dense separately.